tests/movies.spec.js

```diff
@@ -1,45 +1,36 @@
 import { expect, test } from '@jest/globals';
 import request from 'supertest';
 import app from '../app.js';
-
-test('Encanto page shows title of movie', async () => {
-  const response = await request(app).get('/movie/1').expect(200);
-
-  expect(response.text).toMatch('Isle of dogs');
-});
-
-test('Isle of dogs page shows title of movie', async () => {
-  const response = await request(app).get('/movie/2').expect(200);
-
-  expect(response.text).toMatch('Encanto');
-});
-
-test('The Shawshank Redemption page shows title of movie', async () => {
-  const response = await request(app).get('/movie/3').expect(200);
-
-  expect(response.text).toMatch('The Shawshank Redemption');
+import fetch from 'node-fetch';
+
+const url = 'https://plankton-app-xhkom.ondigitalocean.app/api/movies';
+const settings = { method: 'Get' };
+
+let APImapped;
+
+beforeAll(async () => {
+  const response = await fetch(url, settings);
+  const json = await response.json();
+
+  let APIdata = json;
+  APImapped = APIdata.data.map((movies) => {
+    return {
+      id: movies.id,
+      title: movies.attributes.title,
+      intro: movies.attributes.intro,
+      image: movies.attributes.image.url,
+    };
+  });
```

Comment:
This is ok, but there are downsides to both approaches. Hard-coding tests (like you did before) that depend on live data will cause problems if that live data changes. Dynamically loading data in the tests (like you do here) means that there are more things that can go wrong in your tests.

In the end, neither way is recommended. Instead, we usually either mock out the data, or if we want to do a true integration test we do so in a controlled environment where we control the data.

For this assignment however, neither of those solutions were really an option, so both of your solutions are fine!